### PR TITLE
controllers: maintenance mode in ocs-operator

### PR DESCRIPTION
**Pull request by rewantsoni**
If there is a maintenance mode label on any storageConsumer, scale down the rbd mirror daemon deployment

### CODE REVIEW

**Comment by ezio-auditore** [comment]
@rewantsoni I think it would be better if this https://github.com/red-hat-storage/ocs-operator/pull/2838 gets merged before this PR

**Comment by rewantsoni** [comment]
@ezio-auditore I don't see any dependency on the PRs

**Comment by ezio-auditore** [comment]
@rewantsoni I am referring to the test TestCephName which is failing.

File: controllers/storageconsumer/storageconsumer_controller.go [hunk 1]
```diff
@@ -51,6 +53,7 @@ const (
 	StorageProfileLabel           = "ocs.openshift.io/storageprofile"
 	ConsumerUUIDLabel             = "ocs.openshift.io/storageconsumer-uuid"
 	StorageConsumerNameLabel      = "ocs.openshift.io/storageconsumer-name"
+	MaintenanceModeLabel          = "ocs.openshift.io/maintenanceMode"
```
Comment (rchikatw) [inline controllers/storageconsumer/storageconsumer_controller.go:56]: its only used in this file any reason to export this label.

File: controllers/storageconsumer/storageconsumer_controller.go [hunk 2]
```diff
@@ -328,3 +343,42 @@ func addStorageRelatedAnnotations(obj client.Object, storageConsumerName, storag
...
+	err := r.List(r.ctx, rbdMirrorDeployments, client.InNamespace(r.storageConsumer.Namespace), client.MatchingLabels{"app": "rook-ceph-rbd-mirror"}, client.Limit(1))
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if errors.IsNotFound(err) {
+		return nil
+	}
+
+	replicas := int32(1)
+	if enableMaintenanceMode(r.storageConsumerList) {
```
Comment (rchikatw) [inline controllers/storageconsumer/storageconsumer_controller.go:359]: lets fetch rbdMirrorDeployments only if we need to scale down the deployment.

Comment (rewantsoni) [reply controllers/storageconsumer/storageconsumer_controller.go]: I am following this:
1. Fetch rbdMirror, if no rbdMirror found return 
2. Set replicas to 1
3. If enableMaintenanceMode, set replicas to 0
4. Update the Deployment with the desired replica

**Comment by rewantsoni** [comment]
Pivoting with the implementation and it is to be added with https://github.com/red-hat-storage/ocs-operator/pull/2878

**Pull request closed without merging**